src/js/base/runtime.js

```diff
@@ -467,6 +467,19 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
       return thisRuntime.getFieldLoc(obj, field, ["runtime"]);
     }
 
+    function getMaker(obj, makerField, exprLoc, constrLoc) {
+      var maker = isObject(obj) && obj.dict[makerField];
+      if (!isFunction(maker)) {
+        if (thisRuntime.ffi === undefined || thisRuntime.ffi.throwFieldNotFound === undefined) {
```

Comment:
Note https://github.com/brownplt/pyret-lang/issues/1126 (doesn't need fixing for this PR, but notice that the pattern has been copied).

Also, the pattern has been copied incorrectly; I'm pretty sure this should say `throwConstructorSyntaxNonConstructor`, not `throwFieldNotFound`